lib/modules/manager/maven/update.ts

```diff
@@ -78,7 +79,24 @@ export function bumpPackageVersion(
     const startTagPosition = versionNode.startTagPosition;
     const versionPosition = content.indexOf(versionNode.val, startTagPosition);
 
-    const newPomVersion = semver.inc(currentValue, bumpVersion);
+    let newPomVersion: string | null = null;
+    const currentPrereleaseValue = semver.prerelease(currentValue);
+    if (isSnapshot(currentPrereleaseValue)) {
+      let releaseType = bumpVersion;
+      if (!bumpVersion.startsWith('pre')) {
+        releaseType = ('pre' + bumpVersion) as ReleaseType;
+      }
+      newPomVersion = semver.inc(
+        currentValue,
+        releaseType,
+        currentPrereleaseValue?.join('.'),
+        false,
+      );
+    } else if (currentPrereleaseValue) {
+      newPomVersion = semver.inc(currentValue, bumpVersion);
+    } else {
+      newPomVersion = semver.inc(currentValue, bumpVersion, 'SNAPSHOT', false);
+    }
```

Comment:
I don't understand this logic. Today we do a `semver.inc(currentValue, bumpVersion)` for all. Now, you do that only if there's a current prerelease, and if there isn't a prerelease then you do `semver.inc(currentValue, bumpVersion, 'SNAPSHOT', false)`. If this isn't a mistake, it needs explaining in code with comments.

Reply:
I added some comments as requested.